README.md

```diff
@@ -1,57 +1,422 @@
+   - In the *Commonly used Microsoft APIs* section, select **Microsoft Graph**
+   - In the **Delegated permissions** section, select the **User.Read**, **Policy.Read.ConditionalAccess**, **Policy.ReadWrite.ConditionalAccess** in the list. Use the search box if necessary.
+   - `Note: The Graph permission, **Policy.ReadWrite.ConditionalAccess** is required for creating new auth context records by this sample. In production, the permission, **Policy.Read.ConditionalAccess** should be sufficient to read existing values and thus is recommended.`
+   - Select the **Add permissions** button at the bottom.
+
+#### Configure the client app (TodoListClient-authContext-webapp) to use your app registration
+
+Open the project in your IDE (like Visual Studio or Visual Studio Code) to configure the code.
+
+> In the steps below, "ClientID" is the same as "Application ID" or "AppId".
+
+1. Open the `TodoListClient\appsettings.json` file.
+1. Find the key `Domain` and replace the existing value with your Azure AD tenant name.
+1. Find the key `TenantId` and replace the existing value with your Azure AD tenant ID.
+1. Find the key `ClientId` and replace the existing value with the application ID (clientId) of `TodoListClient-authContext-webapp` app copied from the Azure portal.
+1. Find the key `ClientSecret` and replace the existing value with the key you saved during the creation of `TodoListClient-authContext-webapp` copied from the Azure portal.
+
+## Running the sample
+
+> For Visual Studio Users
+>
+> Clean the solution, rebuild the solution, and run it.  You might want to go into the solution properties and set both projects as startup projects, with the service project starting first.
+
+```console
+    cd TodoListClient-authContext-webapp
+    dotnet run
+```
+
+## Explore the sample
+
+### Configure the Auth context
+
+1. We'd first replicate the experience of an admin configuring the auth contexts. For that, browse to `https://localhost:44321` and sign-in using a tenant Admin account. Click on the **Admin** link on the menu.
+
+    ![Overview](./ReadmeFiles/Admin.png)
+2. As a first step, you will ensure that a set of Auth Context is already available in this tenant. Click the **CreateOrFetch** button to check if they exist. If they don't , the code will create three sample auth context entries for you. These three entires are named `Require strong authentication`, `Require compliant devices` and `Require trusted locations`.
+
+    > Note: The Graph permission, **Policy.ReadWrite.ConditionalAccess** is required for creating new auth context records. In production, the permission, **Policy.Read.ConditionalAccess** should be sufficient to read existing values and is thus the only one recommended.
+
+    ![Overview](./ReadmeFiles/Create-Fetch_Click.png)
+
+    Select an operation in the Todo list controller and map it with an `Authentication Context` value. To apply click **SaveOrUpdate**. This updates this mapping in the local app's database.
```

Comment:
It's not clear where the TODO list is because the dropdown is not marked as ToDo but "Web API operation"